.circleci/config.yml

```diff
@@ -0,0 +1,86 @@
+---
+# "Include" for unit tests definition.
+unit_tests: &unit_tests
+  steps:
+    - checkout
+    - run:
+        name: Install modules and dependencies.
+        command: npm install
+    - run:
+        name: Run unit tests.
+        command: npm test
+    - run:
+        name: Submit coverage data to codecov.
+        command: npm run codecov
+        when: always
+
+version: 2.0
+workflows:
+  version: 2
+  tests:
+    jobs:
+      - node4:
+          filters:
+            tags:
+              only: /^v\d+(\.\d+){2}(-.*)?$/
+      - node6:
+          filters:
+            tags:
+              only: /^v\d+(\.\d+){2}(-.*)?$/
+      - node8:
+          filters:
+            tags:
+              only: /^v\d+(\.\d+){2}(-.*)?$/
+      - node9:
+          filters:
+            tags:
+              only: /^v\d+(\.\d+){2}(-.*)?$/
+      - publish_npm:
+          requires:
+            - node4
+            - node6
+            - node8
+            - node9
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v\d+(\.\d+){2}(-.*)?$/
+
+jobs:
+  node4:
+    docker:
+      - image: node:4
+        user: node
+    <<: *unit_tests
+  node6:
+    docker:
+      - image: node:6
+        user: node
+    <<: *unit_tests
+  node8:
+    docker:
+      - image: node:8
+        user: node
+    <<: *unit_tests
+  node9:
+    docker:
+      - image: node:9
+        user: node
+    <<: *unit_tests
+
+  publish_npm:
+    docker:
+      - image: node:8
+        user: node
+    steps:
+      - checkout
+      - run:
+          name: Set npm authentication.
+          command: echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
```

Comment:
If you want npm auto-publish on version tags, you need to add the NPM_TOKEN env var to the CircleCI settings. The token is in valentine.

Reply:
👍 will do this as soon as this lands

Reply:
Which secret in valentine am I looking for?  Can you make sure I have access?

Reply:
It's stored as "google-node-team@npm's NPM_TOKEN". I think you have access. Let me know if you don't.